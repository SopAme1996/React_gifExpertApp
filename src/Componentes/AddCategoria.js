import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategoria = ({ setData }) => {
  const [input, setInput] = useState("");

  const eventoTeclado = (e) => {
    setInput(e.target.value);
  };

  const eventoEnter = (e) => {
    e.preventDefault();

    if (input.trim().length > 2) {
      setData((data) => [input, ...data]);
      setInput("");
    }
  };

  return (
    <form onSubmit={eventoEnter}>
      <input
        type="text"
        value={input}
        onChange={eventoTeclado}
        placeholder="Ingrese su personaje favorito"
      />
    </form>
  );
};

AddCategoria.propTypes = {
  setData: PropTypes.func.isRequired,
};
