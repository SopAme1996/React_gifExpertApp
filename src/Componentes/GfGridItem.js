import React from "react";
import PropTypes from "prop-types";

export const GfGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GfGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GfGridItem;
