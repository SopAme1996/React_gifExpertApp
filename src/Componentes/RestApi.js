import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GfGridItem } from './GfGridItem';
import PropTypes from 'prop-types'


export const RestApi = ({ categoria }) => {

    const {data:imagenes, loading}= useFetchGifs(categoria);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn titulo'>{categoria}</h3>
        <hr className="line"/>
        <div className="animate__animated animate__backInLeft card-grid">

        {loading && <div className='containerLoading'><p>Loading....</p><div className='laoding'></div></div>}
        {
            imagenes.map(img => {
                return(
               <GfGridItem
                   key={img.id}
                   {...img}
              />
              )
            })
                }
        </div>
        </>
    )
}

RestApi.propTypes = {
categoria: PropTypes.string.isRequired
}