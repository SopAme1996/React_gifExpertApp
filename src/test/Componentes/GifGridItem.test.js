import React from 'react';
import { shallow } from 'enzyme';
import { GfGridItem } from '../../Componentes/GfGridItem';

describe('Prueba al componente <GifGridItem/>', () =>{
    const title = 'Un titulo';
    const url = 'https://localhost';
    const wrapper = shallow(<GfGridItem 
        title={title} url={url} />);

    test('Debe de generar los cards', ()=>{
            expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un parrafo con el title', () => {
            const parrafo = wrapper.find('p');
            expect(parrafo.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('el div debe tener  la clase card', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toBe('card');
    })
    
})