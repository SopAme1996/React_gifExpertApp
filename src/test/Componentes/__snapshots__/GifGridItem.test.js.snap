// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba al componente <GifGridItem/> Debe de generar los cards 1`] = `
<div
  className="card"
>
  <img
    alt="Un titulo"
    src="https://localhost"
  />
  <p>
    Un titulo
  </p>
</div>
`;
